wifiphisher/extensions/knownbeacons.py

```diff
@@ -0,0 +1,143 @@
+"""
+Extension that sends a number of known beacons to trigger the AUTO-CONNECT flag.
+"""
+
+import logging
+import time
+from collections import defaultdict
+import wifiphisher.common.constants as constants
+import scapy.layers.dot11 as dot11
+
+logger = logging.getLogger(__name__)
+
+class Knownbeacons(object):
+    """
+    Sends a number of known beacons to trigger the Auto-Connect flag.
+    """
+
+    def __init__(self, shared_data):
+        """
+        Setup the class with all the given arguments
+
+        :param self: A Beacons object
+        :param data: Shared data from main engine
+        :type self: Beacons
+        :type data: dict
+        :return: None
+        :rtype: None
+        """
+
+        self.data = shared_data
+        # store channel to frame list
+        self._packets_to_send = defaultdict(list)
+        self._starttime = time.time()
+        self._msg = []
+        self._full_pkt_list = self._get_known_beacons()
+
+
+    def _get_known_beacons(self):
```

Comment:
The docstring for this function is missing. We should probably add it.

Reply:
Added.